fabric-tool-attributes-v1/src/main/java/net/fabricmc/fabric/impl/tools/ToolManager.java

```diff
@@ -86,28 +88,34 @@ public static void registerBreakByTool(Block block, Tag<Item> tag, int miningLev
 		entry(block).putBreakByTool(tag, miningLevel);
 	}
 
-	private static int getMiningLevel(ItemStack stack) {
-		if (stack.getItem() instanceof ToolItem) {
-			return ((ToolItem) stack.getItem()).getMaterial().getMiningLevel();
+	private static int getMiningLevel(ItemStack stack, ToolActor actor) {
+		if (stack.getItem() instanceof ActableAttributeHolder && !actor.equals(ToolActor.NO_ACTOR)) {
```

Comment:
I suggest checking whether the actor's actual actor is null instead of comparing identity.